### PR TITLE
fixed create eco news bugs #686, 687, 688, 693

### PR DESCRIPTION
**Pull request by Gusvarov**
686: Fields and tags on 'Create news' page remain constantly filled in after 'Back to editing' link is clicked on in the 'Preview' mode
687: It is possible to create news with more than one same tag 
688: User can publish news without selecting any tag
693: 'Publish' button is not disabled when user enters invalid data into 'Source' and/or 'Picture' fields

### CODE REVIEW

File: src/app/component/eco-news/create-news/create-news.component.ts [hunk 1]
```diff
@@ -63,7 +63,9 @@ export class CreateNewsComponent implements OnInit {
 
   public onSourceChange(): void {
     this.createNewsForm.get('source').valueChanges.subscribe(source => {
-      if(source.startsWith('http://') || source.startsWith('https://')) {
+      if(source.startsWith('http://') ||
+        source.startsWith('https://') ||
```
Comment (nabermi2020) [inline src/app/component/eco-news/create-news/create-news.component.ts:67]: Please rewrite it without if/else

File: src/app/component/eco-news/create-news/create-news-cancel/create-news-cancel.component.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import { Component, OnInit } from '@angular/core';
```
Comment (nabermi2020) [inline src/app/component/eco-news/create-news/create-news-cancel/create-news-cancel.component.ts:1]: Bug #686 isn't fixed, please check it one more time

Comment (nabermi2020) [reply src/app/component/eco-news/create-news/create-news-cancel/create-news-cancel.component.ts:1]: Also, please, select 3 tags. After that deselect 2 tags and you'll see that 'Publish' button is disabled. But one tag is still selected. It should be also fixed.

File: src/app/component/eco-news/create-news/create-news-cancel/create-news-cancel.component.ts [hunk 3]
```diff
@@ -1,6 +1,7 @@
...
 import { MatDialogRef } from '@angular/material/dialog';
```
Comment (nabermi2020) [inline src/app/component/eco-news/create-news/create-news-cancel/create-news-cancel.component.ts:2]: Please, resolve conflicts

File: src/app/component/eco-news/create-news/create-news.component.ts (hunk 1 above)
Comment (Gusvarov) [reply src/app/component/eco-news/create-news/create-news.component.ts]: fixed

File: src/app/component/eco-news/create-news/create-news-cancel/create-news-cancel.component.ts (hunk 2 above)
Comment (Gusvarov) [reply src/app/component/eco-news/create-news/create-news-cancel/create-news-cancel.component.ts:1]: fixed

File: src/app/component/eco-news/create-news/create-news-cancel/create-news-cancel.component.ts (hunk 3 above)
Comment (Gusvarov) [reply src/app/component/eco-news/create-news/create-news-cancel/create-news-cancel.component.ts:2]: fixed

File: src/app/service/eco-news/create-eco-news.service.ts [hunk 4]
```diff
@@ -11,6 +11,7 @@ import { FileHandle } from '../../component/eco-news/create-news/create-news-int
 })
 export class CreateEcoNewsService {
   private currentForm: FormGroup;
+  private emptyForm: FormGroup;
```
Comment (KimAT25) [inline src/app/service/eco-news/create-eco-news.service.ts:14]: Please delete unused variable

Comment (Gusvarov) [reply src/app/service/eco-news/create-eco-news.service.ts:14]: fixed

File: src/app/component/eco-news/create-news/create-news-cancel/create-news-cancel.component.ts [hunk 5]
```diff
@@ -1,6 +1,7 @@
...
 import { Router } from '@angular/router';
+import { CreateEcoNewsService } from '../../../../service/eco-news/create-eco-news.service'
```
Comment (nabermi2020) [inline src/app/component/eco-news/create-news/create-news-cancel/create-news-cancel.component.ts:5]: image
Bug 693 is still reproducible with invalid image extension. Please, fix it

Comment (Gusvarov) [reply src/app/component/eco-news/create-news/create-news-cancel/create-news-cancel.component.ts:5]: fixed

**Pull request merged**